packages/serverpod_shared/lib/src/config.dart

```diff
@@ -74,14 +74,30 @@ class ServerpodConfig {
 
     var doc = loadYaml(data);
 
+    /// Get api server setup. This field cannot be null, so if the
+    /// configuration is missing, create a default configuration.
     assert(doc['apiServer'] is Map, 'apiServer is missing in confing');
-    var apiServer = ServerConfig._fromJson(doc['apiServer']);
-
+    Map apiSetup = doc['apiServer'] ?? {};
+    var apiServer = apiSetup.isNotEmpty
+        ? ServerConfig._fromJson(apiSetup)
+        : ServerConfig(
+            port: 8080,
+            publicHost: 'localhost',
+            publicPort: 8080,
+            publicScheme: 'http',
+          );
```

Comment:
Sugggestion: Remove the default. If you specify a file you have to also give a config for the apiServer this makes it explicit and clear for the developer.